lms/djangoapps/certificates/tasks.py

```diff
@@ -18,11 +19,32 @@ class _BaseCertificateTask(PersistOnFailureTask, LoggedTask):  # pylint: disable
     abstract = True
 
 
-@task(base=_BaseCertificateTask)
-def generate_certificate(**kwargs):
+@task(base=_BaseCertificateTask, bind=True, default_retry_delay=30, max_retries=2)
+def generate_certificate(self, **kwargs):
     """
     Generates a certificate for a single user.
+
+    kwargs:
+        - student: The student for whom to generate a certificate.
+        - course_key: The course key for the course that the student is
+            receiving a certificate in.
+        - expected_verification_status: If it is passed in, will be 'approved'.
```

Comment:
If this variable will always be 'approved' if it's passed at all, maybe just pass a boolean?

Reply:
Stylistic questions aside, what happens if I pass in another status? If someone changes the code to pass a status of 'pending' as the expected status, will it generate certs for learners whose verification hasn't been approved?

Reply:
`queue.add_cert()` still checks for a valid status, so this doesn't change anything about the logic of whether or not a cert is generated.  I agree, though, that we should probably just remove any mention of the value of the expected status here, since it could reasonably be something else.  All we care about is that the status that is passed in is the same as what this task can currently read in the DB.

Reply:
Yeah, I think the way it is now, if someone changes the code to pass in another status it would generate the cert. I can change it to the boolean.

Reply:
Cool, sounds good.

Reply:
Oh, sorry, didn't see Alex's comment. I'll keep it as a string instead of a boolean, since the boolean would need to be something like "has_verification_status_changed_to_accepted" which feels clunkier.

Reply:
I changed the comment to not reference the 'accepted' value, though.